CMakeLists.txt

```diff
@@ -41,6 +41,14 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 message("CMAKE_CXX_COMPILER_ID: ${CMAKE_CXX_COMPILER_ID}")
 
+## Debug Usage
+option(ENABLE_CK_DEBUG, "Whether to compile CK as debug message enabled." OFF)
```

Comment:
Maybe we should group all CK's options into one place? I've added recently one for int4. Moreover right now this option is quite ambiguous for me - I'm not sure what it means,  whether it's debug build or what - unless I read it's description. 
I'd propose sth like `CK_DEBUG_LOG`. The "enable` part is not necessary IMHO since it's obvious from the value it's set ie: `-DCK_DEBUG_LOG=ON` clearly means it will be enabled.